### PR TITLE
Approval system for NFT contract

### PR DESCRIPTION
**Pull request by 0x5CE**
Various functions like nft_approve, nft_revoke, nft_revoke_all, nft_is_approved, along with Sandbox tests for them.

### CODE REVIEW

File: packages/contracts/as/NFT/assembly/index.ts [hunk 1]
```diff
@@ -170,6 +177,86 @@ export function nft_transfer(token_id: string, bidder: string): void {
     account_media.set(design.prev_owner, accountMedia)
 }
 
+/* Approval system */
+
+export function nft_approve(token_id: string, account_id: string, msg: string|null = null): void {
+    assert_at_least_one_yocto();
```
Comment (devey) [inline packages/contracts/as/NFT/assembly/index.ts:183]: Why is `assert_one_yocto` not enough here?

**Review by devey — COMMENTED** [review]
At first glance, it looks, great; thanks for starting the implementation of the storage standard as it will help with other parts of the contract too. Let's just add some Sandbox tests then that show how the approval object changes by calling the various methods and then we should be good!

File: packages/contracts/as/NFT/assembly/index.ts (hunk 1 above)
Comment (0x5CE) [reply packages/contracts/as/NFT/assembly/index.ts:183]: I'm not completely sure but I used the Rust example as a reference

Comment (devey) [reply packages/contracts/as/NFT/assembly/index.ts:183]: Got it!

File: packages/contracts/tests/nft.js [hunk 2]
```diff
@@ -140,11 +140,29 @@ async function test() {
...
     const token_owners = await aliceUseContract.nft_tokens_for_owner({account_id: 'alice.test.near'})
 
+    const media = await aliceUseContract.claim_media( { args: {tokenMetadata: {media: ('abc')}},
+        gas: CONTRACT_CLAIM_GAS, amount: CONTRACT_CLAIM_PRICE })
+    
+    // approval tests
+
+    const before_approve = await aliceUseContract.nft_is_approved({token_id: media.id, approved_account_id: 'market.test.near'})
+
+    await aliceUseContract.nft_approve({args: { token_id: media.id, account_id: 'market.test.near' }, gas: CONTRACT_CLAIM_GAS, amount: CONTRACT_CLAIM_PRICE})
```
Comment (devey) [inline packages/contracts/tests/nft.js:150]: Could you write a couple of logs of how media object looks after `approval`, `revoke` @0x5CE ?

**Review by devey — APPROVED** [review]
👍🏼

**Pull request merged**